packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -15,6 +83,12 @@ import 'theme.dart';
 /// will automatically listen to the provided [CupertinoTabBar]'s tap callbacks
 /// to change the active tab.
 ///
+/// A [controller] can be supplied to update the actively displayed tab and manage
+/// subsequent tab changes. If set to null, [CupertinoTabScaffold] will keep
```

Comment:
I wouldn't make one oldController->newController transition of all possible permutations special. Just generically, if a CupertinoTabScaffold is built with a new instance of a CupertinoTabController, that controller instance is used.

Reply:
Then defaulting a null controller to a blank controller isn't a matter of changing the controller instance, it's just a fallback for null.